src/primitives/transaction.h

```diff
@@ -310,12 +310,15 @@ class CTransaction
 
 private:
     /** Memory only. */
+    const bool m_has_witness;
```

Comment:
Not really a fan of adding 8 bytes to every `CTransaction`, though I guess it's not horrible?

Another approach would be to separate out the raw data and compute-logic into one class, and the optimised version with the w/txid cached into a separate class. That looks like: https://github.com/ajtowns/bitcoin/commit/89e3c808927984309a002b2cab1c0a622f5c15f8

Reply:
Personally, I don't think the extra bytes matter that much, so I would lean towards leaving this as is.

Reply:
Really, we're already wasting 16 bytes due to `vin` and `vout` supporting `capacity() != size()` despite both being `const` and never needing to be resized.